code/src/java/pcgen/core/analysis/SkillRankControl.java

```diff
@@ -244,13 +247,6 @@ private static double modRanks2(double rankChange, double curRank,
 					ChooserUtilities.modChoices(sk, new ArrayList<Language>(),
 							new ArrayList<Language>(), aPC, true, null);
 					aPC.setDirty(true);
-					int selectedLanguages = aPC
-							.getSelectCorrectedAssociationCount(sk);
-					int maxLanguages = getTotalRank(aPC, sk).intValue();
-					if (selectedLanguages > maxLanguages)
-					{
-						newRank = curRank;
-					}
```

Comment:
At one point, this code probably operated on side effects - we are sure that is no longer needed?